[zh-cn]sync horizontal-pod-autoscale.md (#51079)
kubernetes/website: content/zh-cn/docs/tasks/run-application/horizontal-pod-autoscale.md

默认值为 5 分钟。
这个配置可以让系统更为平滑地进行缩容操作，从而消除短时间内指标值快速波动产生的影响。

<!--
### Pod readiness and autoscaling metrics

The HorizontalPodAutoscaler (HPA) controller includes two flags that influence how CPU metrics are collected from Pods during startup:
-->
### Pod 准备就绪和自动伸缩指标

> HorizontalPodAutoscaler（HPA）控制器两个标志会影响 Pod 准备就绪和自动伸缩指标的收集：

Reviewer: ```suggestion
HorizontalPodAutoscaler（HPA）控制器有两个标志会影响 Pod 启动期间如何收集其 CPU 指标：
```

<!--
1. `--horizontal-pod-autoscaler-cpu-initialization-period` (default: 5 minutes)

  This defines the time window after a Pod starts during which its **CPU usage is ignored** unless:
    - The Pod is in a `Ready` state **and**
    - The metric sample was taken entirely during the period it was `Ready`.
-->
1. `--horizontal-pod-autoscaler-cpu-initialization-period`（默认：5 分钟）

>   这定义了 Pod 启动后的一段时间，在此期间内其 **CPU 使用率被忽略**，除非：

Reviewer: ```suggestion
   此标志所定义的是 Pod 启动后的一个时间窗口，在此期间内其 **CPU 使用率被忽略**，除非：
```

    - Pod 处于 `Ready` 状态**且**
    - 指标样本完全是在它处于 `Ready` 状态期间采集的。

>   <!--
>   This flag helps **exclude misleading high CPU usage** from initializing Pods (e.g., Java apps warming up) in HPA scaling decisions.
>   -->
> 
>   此标志有助于**排除初始化 Pod 中的误导性高 CPU 使用率**
>   （例如，Java 应用程序预热）对 HPA 扩缩决策的影响。

Reviewer: ```suggestion
   <!--
   This flag helps **exclude misleading high CPU usage** from initializing Pods (e.g., Java apps warming up) in HPA scaling decisions.
  -->

   此标志有助于**去除初始化 Pod 过程中的误导性高 CPU 使用率值**
   （例如，Java 应用程序预热）对 HPA 扩缩决策的影响。
```

<!--
2. `--horizontal-pod-autoscaler-initial-readiness-delay` (default: 30 seconds)

  This defines a short delay period after a Pod starts during which the HPA controller treats Pods that are currently `Unready` as still initializing, **even if they have previously transitioned to `Ready` briefly**.
-->
2. `--horizontal-pod-autoscaler-initial-readiness-delay`（默认：30 秒）

>   这定义了一个短暂的延迟期，在 Pod 启动后，HPA 控制器将当前为 `Unready`
>   的 Pod 视为仍在初始化中，**即使它们之前曾短暂转变为 `Ready`**。

Reviewer: ```suggestion
   此标志定义的一个短暂的延迟期，在 Pod 启动后的这一延迟期内，HPA 控制器将当前为 `Unready`
   的 Pod 视为仍在初始化中，**即使它们之前曾短暂转变为 `Ready`**。
```

>   <!--
>   It is designed to:
>     - Avoid including Pods that rapidly fluctuate between `Ready` and `Unready` during startup.
>     - Ensure stability in the initial readiness signal before HPA considers their metrics valid.
>   -->
> 
>   其设计目的是：
>     - 排除包含在启动期间快速在 `Ready` 和 `Unready` 之间波动的 Pod。
>     - 确保在 HPA 认为它们的指标有效之前，初始就绪信号的稳定性。

Reviewer: ```suggestion
   <!--
   It is designed to:
   - Avoid including Pods that rapidly fluctuate between `Ready` and `Unready` during startup.
   - Ensure stability in the initial readiness signal before HPA considers their metrics valid.
   -->
   其设计目的是：
   - 避免包含在启动期间快速在 `Ready` 和 `Unready` 之间波动的 Pod。
   - 确保在 HPA 认为它们的指标有效之前，初始就绪信号的稳定性。
```

<!--
**Key behaviors:**
- If a Pod is `Ready` and remains `Ready`, it can be counted as contributing metrics even within the delay.
- If a Pod rapidly toggles between `Ready` and `Unready`, metrics are ignored until it’s considered stably `Ready`.
-->
**关键行为：**

- 如果一个 Pod 是 `Ready` 状态并且保持 `Ready`，即使在延迟期间，

>   它也可以被计为贡献指标。

Reviewer: ```suggestion
  它也可以被视为贡献指标。
```

- 如果一个 Pod 在 `Ready` 和 `Unready` 之间快速切换，那么它的指标将被忽略，
  直到它被认为稳定在 `Ready` 状态。

<!--
#### Best Practice:
If your Pod has a startup phase with high CPU usage, configure both:
- `--horizontal-pod-autoscaler-cpu-initialization-period` to **cover the startup duration**.
- Ensure your **readinessProbe** only reports `Ready` **after the CPU spike subsides**, using `initialDelaySeconds`.
-->
#### 最佳实践

> 如果你的 Pod 有一个高 CPU 使用率的启动阶段，配置以下两项：

Reviewer: ```suggestion
如果你的 Pod 有一个 CPU 使用率较高的启动阶段，可以配置以下两项：
```

- `--horizontal-pod-autoscaler-cpu-initialization-period`
  以**覆盖启动持续时间**。

> - 确保你的 **readinessProbe** 在 CPU 峰值消退后才报告 `Ready`，
>   通过使用 `initialDelaySeconds`。

Reviewer: ```suggestion
- 通过使用 `initialDelaySeconds`，确保你的 **readinessProbe** 在 CPU 峰值消退后才报告 `Ready`。
```

<!--
This avoids scaling based on temporary spikes that do not reflect long-term workload needs.
-->

> 这可以避免基于临时峰值的缩放，这些峰值不反映长时间工作负载需求。

Reviewer: ```suggestion
这样做可以避免基于临时峰值的缩放动作，这些峰值不反映长时间工作负载需求。
```
